src/components/base/outline-google-map/outline-google-map.ts

```diff
@@ -0,0 +1,248 @@
+import { LitElement, html, CSSResultGroup } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import { GoogleMapsApi } from './src/google-maps-api';
+import './src/google-maps-api';
+import { OutlineMapMarker } from './src/outline-map-marker';
+import { MapSelector } from './src/map-selector';
+import './src/map-selector';
+import componentStyles from './outline-google-map.css.lit';
+
+export type MapType = 'roadmap' | 'hybrid' | 'satellite' | 'terrain';
+export type GestureType = 'auto' | 'cooperative' | 'greedy' | 'none';
+
+@customElement('outline-google-map')
+export class OutlineGoogleMap extends LitElement {
+  static styles: CSSResultGroup = [componentStyles];
+  /**
+   * A Maps API key. To obtain an API key, see https://developers.google.com/maps/documentation/javascript/tutorial#api_key.
+   */
+  @property({ type: String })
+  apiKey = '';
+
+  /**
+   * Version of the Google Maps API to use.
+   */
+  @property({ type: String })
+  version = '3.48';
+
+  /**
+   * If set, custom styles can be applied to the map.
+   * For style documentation see https://developers.google.com/maps/documentation/javascript/reference#MapTypeStyle
+   */
+  @property({ type: Object })
+  styles: google.maps.MapTypeStyle[] = [];
```

Comment:
How does this work with an HTML consumer? Does this only work with a JS framework?

When we've extended styles, we've extended the component, but there may be better ways.

There is also a way to process attributes and convert them to properties that might work if you want to pass in JSON. If it doesn't do that already.

Reply:
https://lit.dev/docs/components/properties/#conversion-converter